scripts/zkai_d128_value_adapter_policy_frontier_gate.py

```
@@ -0,0 +1,648 @@
+def best_global_affine(base: list[int], target: list[int]) -> list[int]:
+    best: tuple[int, int, int, int, list[int]] | None = None
+    for scale in range(-64, 65):
+        for bias in range(-256, 257):
+            values = [scale * value + bias for value in base]
+            mismatches = sum(candidate != target_value for candidate, target_value in zip(values, target, strict=True))
+            total_abs_error = sum(abs(candidate - target_value) for candidate, target_value in zip(values, target, strict=True))
+            item = (mismatches, total_abs_error, scale, bias, values)
+            if best is None or item[:4] < best[:4]:
+                best = item
+    if best is None:
+        raise ValueAdapterPolicyFrontierError("failed global affine search")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)